tests/core/pyspec/eth2spec/test/phase0/fork_choice/test_on_block.py

```diff
@@ -200,21 +203,31 @@ def test_on_block_finalized_skip_slots(spec, state):
     on_tick_and_append_step(spec, store, current_time, test_steps)
     assert store.time == current_time
 
-    # Create a finalized chain
-    for _ in range(4):
+    # Fill epoch 0 and the first slot of epoch 1
+    state, store, _ = yield from apply_next_slots_with_attestations(
+        spec, state, store, spec.SLOTS_PER_EPOCH, True, False, test_steps)
+
+    # Skip the rest slots of epoch 1 and the first slot of epoch 2
+    next_slots(spec, state, spec.SLOTS_PER_EPOCH)
+
+    # The state after the skipped slots
+    target_state = state.copy()
+
+    # Fill epoch 3 and 4
+    for _ in range(2):
         state, store, _ = yield from apply_next_epoch_with_attestations(
-            spec, state, store, True, False, test_steps=test_steps)
-    assert store.finalized_checkpoint.epoch == 2
+            spec, state, store, True, True, test_steps=test_steps)
 
-    # Another chain
-    another_state = store.block_states[store.finalized_checkpoint.root].copy()
-    # Build block that includes the skipped slots up to finality in chain
-    block = build_empty_block(spec,
-                              another_state,
-                              spec.compute_start_slot_at_epoch(store.finalized_checkpoint.epoch) + 2)
-    block.body.graffiti = b'\x12' * 32
-    signed_block = state_transition_and_sign_block(spec, another_state, block)
+    # Now we get finalized epoch 1, where compute_start_slot_at_epoch(1) is a skipped slot
+    assert state.finalized_checkpoint.epoch == store.finalized_checkpoint.epoch == 2
+    assert store.finalized_checkpoint.root == spec.get_block_root(state, 1) == spec.get_block_root(state, 2)
+    assert state.current_justified_checkpoint.epoch == store.justified_checkpoint.epoch == 3
+    assert store.justified_checkpoint == state.current_justified_checkpoint
 
+    # Now build a block at later slot than finalized *epoch*
+    # Includes finalized block in chain and the skipped slots
+    block = build_empty_block_for_next_slot(spec, target_state)
```

Comment:
This block is valid wrt the finalized epoch (including the skip slots), but this block isn't the head, right?

`apply_next_epoch_with_attestations` for epoch 3 and 4 will have created blocks of much higher weight and win out

So -- Does adding this block change anything? or does it catch a regression in the prior fork choice implementations?

Reply:
and if `block` is supposed to be the head? can we check that it is against a dynamic call to `get_head()`?

Reply:
Ah, I see. It's just checking in `on_block` can even be run anymore which would yield `valid: False` if it didn't work

Reply:
> Ah, I see. It's just checking in on_block can even be run anymore which would yield valid: False if it didn't work

Right, it is for comparing with `test_on_block_finalized_skip_slots_not_in_skip_chain` (valid: False) test case.